lib/config/config-validator.js

```diff
@@ -194,18 +195,16 @@ function formatErrors(errors) {
  * for each unique file path, but repeated invocations with the same file path have no effect.
  * No warnings are emitted if the `--no-deprecation` or `--no-warnings` Node runtime flags are active.
  * @param {string} source The name of the configuration source to report the warning for.
+ * @param {string} message The warning message to show.
  * @returns {void}
  */
-const emitEcmaFeaturesWarning = lodash.memoize(source => {
-
-    /*
-     * util.deprecate seems to be the only way to emit a warning in Node 4.x while respecting the --no-warnings flag.
-     * (In Node 6+, process.emitWarning could be used instead.)
-     */
-    util.deprecate(
-        () => {},
-        `[eslint] The 'ecmaFeatures' config file property is deprecated, and has no effect. (found in ${source})`
-    )();
+const emitDeprecationWarning = lodash.memoize((source, message) => {
+    const rel = path.relative(process.cwd(), source);
+
+    process.emitWarning(
+        `${message} (found in "${rel}")`,
+        { code: "ESLint", type: "DeprecationWarning" }
```

Comment:
I think the `code` property of a warning is generally supposed to be unique for a given type of warning, so that tools can filter warnings reliably even if the message changes. Maybe it would be better to do something like `{ code: 'ESLINT_EXPERIMENTAL_OBJECT_REST_SPREAD' }` instead, to make sure it's distinguished from other warnings.

Reply:
I see. Today I leaned the meaning of the `code` property.

However, I feel reluctant to make `code` longer since the string will be printed for users. To be clear, Node.js prints this message such as:

```
(node:10736) [ESLint] DeprecationWarning: The 'parserOptions.ecmaFeatures.experimentalObjectRestSpread' option is deprecated. Use 'parserOptions.ecmaVersion' instead. (found in "tests\fixtures\config-file\experimental-object-rest-spread\extends\common.yml")
```

The first `[ESLint]` is the `code`.

Reply:
Maybe we could make it something like `ESLINT_EORS`?